Yearn-agents/Large-Deposit-Withdraw/src/agent.ts

```diff
@@ -0,0 +1,56 @@
+import { 
+  Finding, 
+  HandleTransaction, 
+  TransactionEvent, 
+  getJsonRpcUrl,
+} from 'forta-agent';
+import deposit from './deposit';
+import withdraw from './withdraw';
+import DataFetcher from './fetcher';
+import BigNumber from 'bignumber.js';
+import Web3 from 'web3';
+
+const PERCENT: number = 40 // % that define what large means
+const YEARN_PROVIDER: string = '0x437758d475f70249e03eda6be23684ad1fc375f0';  
+
+export const provideHandleTransaction = (
+  provider: string, 
+  percent: number, 
+  fetcher: DataFetcher,
+): HandleTransaction => 
+  async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const vaults: string[] = await fetcher.getVaults(provider, txEvent.blockNumber);
+    const depositLimit: BigNumber[] = await Promise.all(
+      vaults.map((v: string) => fetcher.getMaxDeposit(v, txEvent.blockNumber - 1)),
+    );
+    const withdrawLimit: BigNumber[] = await Promise.all(
+      vaults.map((v: string) => fetcher.getMaxWithdraw(v, txEvent.blockNumber - 1)),
+    );
+
+    const handlers: HandleTransaction[] = [];
+    for(let i = 0; i < vaults.length; ++i){
+      handlers.push(deposit.provideLargeDepositDetector(
+        vaults[i], 
+        depositLimit[i].multipliedBy(percent).div(100),
+      ));
+    }
+    for(let i = 0; i < vaults.length; ++i){
```

Comment:
Sure I can, I just did two loops in order to group the two types of alerts instead of mixing them all. This doesn't affect the time complexity of then it should be fine.